java-squid/src/main/java/org/sonar/java/model/declaration/AnnotatedDimension.java

```diff
@@ -0,0 +1,78 @@
+/*
+ * SonarQube Java
+ * Copyright (C) 2012 SonarSource
+ * dev@sonar.codehaus.org
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.model.declaration;
+
+import com.google.common.collect.ImmutableList;
+import com.sonar.sslr.api.AstNode;
+import org.sonar.java.model.InternalSyntaxToken;
+import org.sonar.plugins.java.api.tree.SyntaxToken;
+
+import java.util.List;
+
+public class AnnotatedDimension {
+
+  private final List<AnnotationTreeImpl> annotations;
+  private final SyntaxToken openBracketToken;
+  private final SyntaxToken closeBracketToken;
+  private final SyntaxToken ellipsisToken;
+
+  public AnnotatedDimension(List<AnnotationTreeImpl> annotations, SyntaxToken openBracketToken, SyntaxToken closeBracketToken) {
+    this.annotations = annotations;
+    this.openBracketToken = openBracketToken;
+    this.closeBracketToken = closeBracketToken;
+    this.ellipsisToken = null;
+  }
+
+  public AnnotatedDimension(List<AnnotationTreeImpl> annotations, SyntaxToken ellipsisToken) {
+    this.annotations = annotations;
+    this.openBracketToken = null;
+    this.closeBracketToken = null;
+    this.ellipsisToken = ellipsisToken;
+  }
+
+  public List<AnnotationTreeImpl> annotations() {
+    return annotations;
+  }
+
+  public SyntaxToken openBracketToken() {
+    return openBracketToken;
+  }
+
+  public SyntaxToken closeBracketToken() {
+    return closeBracketToken;
+  }
+
+  public SyntaxToken ellipsisToken() {
+    return ellipsisToken;
+  }
+
+  public List<AstNode> getAstNodes() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) org/sonar/java/model/declaration/AnnotatedDimension must not use com/sonar/sslr/api/AstNode [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AAstNode_SSLR_class_should_not_be_used)